cmd/thanos/sidecar.go

```diff
@@ -222,16 +232,18 @@ func runSidecar(
 
 		g.Add(func() error {
 			level.Info(logger).Log("msg", "Listening for StoreAPI gRPC", "address", grpcBindAddr)
+			readinessProber.SetReady()
 			return errors.Wrap(s.Serve(l), "serve gRPC")
-		}, func(error) {
+		}, func(err error) {
+			readinessProber.SetNotReady(err)
```

Comment:
I mean, setting in one function like this is enough

Reply:
also healthyness

Reply:
I'm not sure if I understand correctly what exactly do you mean by the `setting in one function`.
You mean dropping at all changing the ready status because of prom ext labels fetch?

Also not sure about the readiness vs healthyness. The sidecar in this case could be still shipping some buckets to OS so killing it just because the gRPC interface has malfunction could be too harsh?